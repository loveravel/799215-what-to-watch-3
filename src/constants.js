export const DEFAULT_GENRE = `All genres`;

export const EMAIL_REGEX = RegExp(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);

export const BASE_URL = `https://htmlacademy-react-3.appspot.com`;

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  MOVIE: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  VIDEO_PLAYER: `/player/:id`,
};

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const FilmsCount = {
  STEP: 8,
  INITIAL_MAIN: 8,
  INITIAL_SIMILAR: 4,
};

export const RatingDescription = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very Good`,
  AWESOME: `Awesome`,
};

export const ErrorText = {
  EMAIL: `Please enter a valid email address.`,
  PASSWORD: `Please enter a valid password.`
};

export const InputName = {
  EMAIL: `email`,
  PASSWORD: `password`,
};

export const FavoriteStatus = {
  YES: 1,
  NO: 0,
};

export const CommentLength = {
  MIN: 50,
  MAX: 400,
};
