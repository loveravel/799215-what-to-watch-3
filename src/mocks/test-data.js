export const films = [
  {
    id: 1,
    name: `Movie 1`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: true
  },
  {
    id: 2,
    name: `Movie 2`,
    genre: ``,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideoLink: ``,
    video: ``,
    isFavorite: false
  },
  {
    id: 3,
    name: `Movie 3`,
    genre: `Drama`,
    released: 2019,
    backgroundImage: ``,
    backgroundColor: ``,
    previewImage: ``,
    posterImage: ``,
    description: ``,
    rating: 4.5,
    scoresCount: 300,
    runTime: 3,
    director: ``,
    starring: [],
    previewVideo: ``,
    video: ``,
    isFavorite: false
  },
];

export const movie = {
  id: 1,
  name: `Movie 1`,
  genre: ``,
  released: 2019,
  backgroundImage: ``,
  backgroundColor: ``,
  previewImage: ``,
  posterImage: ``,
  description: ``,
  rating: 4.5,
  scoresCount: 300,
  runTime: 3,
  director: ``,
  starring: [],
  previewVideo: ``,
  video: ``,
  isFavorite: false
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 1,
      name: ``
    },
    rating: 5.5,
    comment: ``,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const userData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};
