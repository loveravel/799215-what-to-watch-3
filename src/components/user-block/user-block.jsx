import React from 'react';
import PropTypes from "prop-types";
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import {getAuthorizationStatus, getUserData} from "../../reducer/user/selectors.js";

import {BASE_URL} from "../../constants.js";
import history from "../../history.js";

const UserBlock = (props) => {
  const {userData, authorized} = props;
  return (
    <div className="user-block">
      {
        !authorized
          ? (<Link to="/login" className="user-block__link">Sign in</Link>)
          : (
            <div className="user-block__avatar">
              <img
                src={`${BASE_URL}${userData.avatarUrl}`}
                alt="User avatar"
                width="63"
                height="63"
                onClick={() => history.push(`/mylist`)}
              />
            </div>
          )
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    userData: getUserData(state),
    authorized: getAuthorizationStatus(state)
  };
};

UserBlock.propTypes = {
  authorized: PropTypes.bool,
  userData: PropTypes.object,
};

export {UserBlock};

export default connect(mapStateToProps)(UserBlock);
