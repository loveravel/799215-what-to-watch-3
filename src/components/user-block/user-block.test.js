import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";

import {userData} from "../../mocks/test-data.js";
import {AuthorizationStatus} from "../../constants.js";
import history from "../../history.js";

import {UserBlock} from "./user-block.jsx";

it(`<UserBlock/> is rendered correctly with authorization`, () => {
  const props = {
    userData,
    authorized: AuthorizationStatus.AUTH,
  };

  const tree = renderer
    .create(
        <Router history={history}>
          <UserBlock {...props}/>
        </Router>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<UserBlock/> is rendered correctly without authorization`, () => {
  const props = {
    userData,
    authorized: AuthorizationStatus.NO_AUTH,
  };

  const tree = renderer
  .create(
      <Router history={history}>
        <UserBlock {...props}/>
      </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
