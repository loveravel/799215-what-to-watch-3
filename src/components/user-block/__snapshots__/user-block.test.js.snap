// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBlock/> is rendered correctly with authorization 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      onClick={[Function]}
      src="https://htmlacademy-react-3.appspot.com"
      width="63"
    />
  </div>
</div>
`;

exports[`<UserBlock/> is rendered correctly without authorization 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
